source/postprocess/particle_crystal_preferred_orientation.cc

```diff
@@ -0,0 +1,1036 @@
+          // We also provide a warning about this fact. There are places where
+          // this fails *on every node*, so we will get a lot of warning messages
+          // into the output; in these cases, just writing multiple pieces to
+          // std::cerr will produce an unreadable mass of text; rather, first
+          // assemble the error message completely, and then output it atomically
+          if (tmp_file_desc == -1)
+            {
+              const std::string x = ("***** WARNING: could not create temporary file <"
+                                     +
+                                     tmp_filename
+                                     +
+                                     ">, will output directly to final location. This may negatively "
+                                     "affect performance. (On processor "
+                                     + Utilities::int_to_string(Utilities::MPI::this_mpi_process (MPI_COMM_WORLD))
+                                     + ".)\n");
+
+              std::cerr << x << std::flush;
+
+              tmp_filename = filename;
+            }
+          else
+            close(tmp_file_desc);
+        }
+
+      std::ofstream out(tmp_filename.c_str(), std::ofstream::binary);
+
+      AssertThrow (out, ExcMessage(std::string("Trying to write to file <") +
+                                   filename +
+                                   ">, but the file can't be opened!"))
+
+      // now write and then move the tmp file to its final destination
+      // if necessary
+      if (compress_contents)
+        {
+          namespace bio = boost::iostreams;
+
+          std::stringstream compressed;
+          std::stringstream origin(*file_contents);
+
+          bio::filtering_streambuf<bio::input> compress_out;
+          compress_out.push(bio::zlib_compressor());
+          compress_out.push(origin);
+          bio::copy(compress_out, out);
+        }
+      else
+        {
+          out << *file_contents;
+        }
+
+      out.close ();
+
+      if (tmp_filename != filename)
+        {
+          std::string command = std::string("mv ") + tmp_filename + " " + filename;
+          int error = system(command.c_str());
+
+          AssertThrow(error == 0,
+                      ExcMessage("Could not move " + tmp_filename + " to "
+                                 + filename + ". On processor "
+                                 + Utilities::int_to_string(Utilities::MPI::this_mpi_process (MPI_COMM_WORLD)) + "."));
+        }
+
+      // destroy the pointer to the data we needed to write
+      delete file_contents;
+    }
+
+    template <int dim>
+    std::pair<std::string,std::string>
+    ParticleCrystalPreferredOrientation<dim>::execute (TableHandler &statistics)
+    {
+
+      const Particle::Property::Manager<dim> &manager = this->get_particle_world().get_property_manager();
+
+      bool hexagonal_plugin_exists = manager.plugin_name_exists("decompose elastic matrix");
```

Comment:
Is this plugin part of another open PR?
```suggestion
      const bool hexagonal_plugin_exists = manager.plugin_name_exists("decompose elastic matrix");
```

Reply:
That pull request is not open yet since it depends on #4987, which depends on this pull request. I will remove it for now and add it (and the related code) back in that pull request.